rllib/utils/minibatch_utils.py

```diff
@@ -187,6 +187,37 @@ def __iter__(self):
         yield self._batch
 
 
+@DeveloperAPI
+class MiniBatchRayDataIterator:
+    def __init__(self, *, iterator, finalize_fn, num_iters, **kwargs):
+        self._iterator = iterator
+        self._finalize_fn = finalize_fn
+        self._num_iters = num_iters
+        self._kwargs = kwargs
+
+    def __iter__(self):
+        iter = 0
+        while self._num_iters is None or iter < self._num_iters:
```

Comment:
I have to see, if this will still work with the newest changes in one of my branches. The iterator needs to be pausable. Generators are possible and this is a generator, but an instance of this class will be assigned to `batch_iter` in the `Learner.update`. For now this works.